src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,107 @@
+package duke.parser;
+
+import duke.DukeException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.TaskList;
+import duke.task.Todo;
+
+import java.util.List;
+
+/**
+ * Represents a parser that interprets the user's inputs into commands.
+ */
+public class Parser {
+    static String byeMsg = "Bye. Hope to see you again soon!";
+    static String listMsg = "Here are the tasks in your list:";
+    static String doneMsg = "Nice! I've marked this duke.task as done:";
+    static String todoMsg = "Got it. I've added this duke.task:";
+    static String deleteMsg = "Noted. I've removed this duke.task:";
+    static String findMsg = "Here are the matching tasks in your list:";
+
+    /**
+     * A method to convert the user's input into a command.
+     *
+     * @param userInput the user's input given as a string
+     * @param taskList  the list containing the tasks
+     * @throws DukeException if the inputs are not expected
+     */
+    public static void parse(String userInput, TaskList taskList) throws DukeException {
```

Comment:
The parser function is a bit long. Can seperate the cases into diffrent methods.